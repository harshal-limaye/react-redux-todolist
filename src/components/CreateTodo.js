import React from 'react';
import { connect } from 'react-redux';
import { addToDo } from '../actions/todos';

class CreateTodo extends React.Component {
    render () {
        return (
            <div className="todo-create">
                <input type="text" onKeyDown={this.handleEnter} />
            </div>
        );
    }

    handleEnter = (e) => {
        if (e.keyCode === 13 && e.target.value && e.target.value.trim() !== '') {
            this.props.dispatch(addToDo(e.target.value));
            e.target.value = '';
        }
    }
}

export default connect()(CreateTodo);