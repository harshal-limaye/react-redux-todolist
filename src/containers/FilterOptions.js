import { connect } from 'react-redux';
import { setVisibilityFilter } from '../actions/visibility';
import Filter from '../components/Filter';

const mapStateToProps = state => ({ visibility: state.visibility || '' });

const mapDispatchToProps = dispatch => ({ 
    setFilter: filter => dispatch(setVisibilityFilter(filter)) 
});

export default connect(mapStateToProps, mapDispatchToProps)(Filter);

