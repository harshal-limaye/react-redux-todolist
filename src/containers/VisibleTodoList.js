import { connect } from 'react-redux';
import { completeToDO, deleteToDO } from '../actions/todos';
import TodoList from '../components/TodoList';


const getFilteredTodos = ({ todos, visibility }) => {
    switch (visibility) {
        case 'ALL':
            return todos;
        case 'ACTIVE':
            return todos.filter(t => !t.completed);
        case 'COMPLETED':
            return todos.filter(t => t.completed);
        default:
            throw new Error('Unknown filter: ' + visibility);
    }
}

const mapStateToProps = (state) => ({
    todos: getFilteredTodos(state) || []
});

const mapDispatchToProps = (dispatch) => ({
    complete: id => dispatch(completeToDO(id)),
    delete: id => dispatch(deleteToDO(id)),
});

export default connect(mapStateToProps, mapDispatchToProps)(TodoList);