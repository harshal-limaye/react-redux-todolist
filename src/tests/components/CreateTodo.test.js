import React from 'react';
import { shallow } from 'enzyme';
import { CreateTodo } from '../../containers/CreateTodo';

describe('testing create todo component', () => {

    test('should render create todo properly', () => {
        const wrapper = shallow(<CreateTodo />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should handle keydown event', () => {
        const e = {
            keyCode: 13,
            target: {
                value: 'sample'
            }
        };
        const dispatch = jest.fn();
        const wrapper = shallow(<CreateTodo dispatch={dispatch} />);
        wrapper.instance().handleEnter(e);
        expect(e.target.value).toBe('');
    });

    test('should not handle keydown event if enter not pressed', () => {
        const e = {
            keyCode: 49,
            target: {
                value: 'sample'
            }
        };
        const dispatch = jest.fn();
        const wrapper = shallow(<CreateTodo dispatch={dispatch} />);
        wrapper.instance().handleEnter(e);
        expect(e.target.value).toBe('sample');
    });
});