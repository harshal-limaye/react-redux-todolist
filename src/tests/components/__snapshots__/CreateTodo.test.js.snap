// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing create todo component should render create todo properly 1`] = `
<div
  className="todo-create"
>
  <input
    onKeyDown={[Function]}
    placeholder="Task..."
    type="text"
  />
</div>
`;
